math-scala/src/main/scala/org/apache/mahout/math/algorithms/preprocessing/AsFactor.scala

```diff
@@ -0,0 +1,117 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements. See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership. The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.preprocessing
+
+import org.apache.mahout.math.{Vector => MahoutVector}
+
+import collection._
+import JavaConversions._
+import org.apache.mahout.math._
+import org.apache.mahout.math.drm.{drmBroadcast, _}
+import org.apache.mahout.math.drm.RLikeDrmOps._
+import org.apache.mahout.math.scalabindings.{dvec, _}
+import org.apache.mahout.math.scalabindings.RLikeOps._
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+import scala.reflect.ClassTag
+
+class AsFactor extends PreprocessorModelFactory {
+
+  def fit[K](input: DrmLike[K]) = {
+
+    import org.apache.mahout.math.function.VectorFunction
+    val factorMap = input.allreduceBlock(
+      { case (keys, block) =>
+        // someday we'll replace this with block.max: Vector
+        // or better yet- block.distinct
+        dense(block.aggregateColumns( new VectorFunction {
+            def apply(f: Vector): Double = f.max
+        }))
+      })(0, ::)
+    new AsFactorModel(factorMap.sum.toInt, factorMap)
```

Comment:
style nitpicking. since this really does not overload a trait-signed method, it would be best readable if the return type is explicitly stipulated in the declaration.
```
def fit[K](input: DrmLike[K]):MyModelType = {
```